### DISCUSSION TITLE
View Distance for Separate Layers

### DISCUSSION THREAD

**Discussion by neverwinter-nights** [o3de/o3de#15858] [Help Wanted]
Hello.

I would like to separate game objects into several layers, like in Photoshop or similar editor.
The first layer would contain terrain mesh – ground, water, mountains, sands and swamps.
The second layer would contain optional terrain objects like small bushes, grass, small rocks.
The third layer would contain various constructs, such as bridges, roads, road signs, houses, castles, pieces of wall, etc.
The fourth layer would contain characters and other living creatures, such as animals, humans, elves, orcs and so on.

And the key feature I want to have is to able to control maximum visible distance for all objects of each layer separately and dynamically. For example, in daylight the view distance of 4-th layer will be greater than at night. Some layers would have greater view distance radius than other layers.

Is it possible in O3DE and if yes, what is the best way to implement such a thing ?

Thank you.

**Comment by vault-thirteen** [comment]
Does any of developers or users of this 3D engine have any knowledge about the possibility of the described functionality ?

**Comment by ceceomer** [comment]
There is no engine support for this fuature because it would be useless. You should calculate your view side then enable/disable them.
